lib/ParseSIL/ParseSIL.cpp

```diff
@@ -885,6 +887,158 @@ void SILParser::convertRequirements(SILFunction *F,
+            value = SymbolicValue::getFloat(floatValue, allocator);
+            return false;
+          }
+          case 64: {
+            APFloat floatValue(APFloat::IEEEdouble(), bits);
+            value = SymbolicValue::getFloat(floatValue, allocator);
+            return false;
+          }
+          default:
+            P.diagnose(datatypeToken, diag::sil_const_expected_fp_datatype);
+            return true;
+        }
+      }
+      // Handle decimal case.
+      bool negative = false;
+      if (P.Tok.isAnyOperator() && P.Tok.getText() == "-") {
+        negative = true;
+        P.consumeToken();
+      }
+      if (!P.Tok.is(tok::floating_literal)) {
+        P.diagnose(P.Tok, diag::sil_const_expected_fp_value);
+        return true;
+      }
+      double tmpValue;
+      P.Tok.getText().getAsDouble(tmpValue);
+      P.consumeToken(tok::floating_literal);
+
+      APFloat floatValue((float) 0);
+      switch (width) {
+      case 32:
+        floatValue = APFloat((float) tmpValue);
+        break;
+      case 64:
+        floatValue = APFloat(tmpValue);
+        break;
+      default:
+        P.diagnose(datatypeToken, diag::sil_const_expected_fp_datatype);
+        return true;
+      }
+      if (negative)
+        floatValue.changeSign();
+      value = SymbolicValue::getFloat(floatValue, allocator);
+      return false;
+    }
+  }
+  // Handle string literals.
+  if (P.Tok.is(tok::string_literal)) {
+    // TODO: Uncomment when `getStringValue` is implemented.
+    // StringRef rawString = P.Tok.getText().drop_front().drop_back();
+    // value = SymbolicValue::getStringValue(rawString, allocator);
+    // return false;
+    P.diagnose(P.Tok, diag::sil_graph_op_unhandled_attr_value);
+    return true;
+  }
+  // Handle metatypes (the instance type is parsed).
+  if (P.Tok.is(tok::sil_dollar)) {
+    SILType temp;
+    if (SP.parseSILType(temp))
+      return true;
+    auto metatype = CanMetatypeType::get(temp.getSwiftRValueType());
+    value = SymbolicValue::getMetatype(metatype);
+    return false;
+  }
+  // Handle aggregate literals.
+  if (P.Tok.is(tok::l_square)) {
+    SourceLoc lSquareLoc = P.consumeToken(tok::l_square);
+    SourceLoc rSquareLoc;
+
+    SmallVector<SymbolicValue, 8> elements;
+    ParserStatus status =
+      P.parseList(tok::r_square, lSquareLoc, rSquareLoc,
+                  /*AllowSepAfterLast*/ false,
+                  diag::sil_const_aggregate_expected_rsquare,
+                  SyntaxKind::Unknown,
+                  [&]() -> ParserStatus {
+      SymbolicValue element;
+      if (parseSymbolicValue(element, SP, B))
+        return makeParserError();
+      elements.push_back(element);
+      return makeParserSuccess();
+    });
+    if (status.isError())
+      return true;
+    value = SymbolicValue::getAggregate(elements, allocator);
+    return false;
+  }
+  P.diagnose(P.Tok, diag::sil_graph_op_expected_attr_value);
+  return true;
+};
+
 /// SWIFT_ENABLE_TENSORFLOW
```

Comment:
this tag can be consolidated with the above code?

Reply:
I would prefer to keep multiple tags so that each function is marked clearly.